src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.vmware.vipclient.i18n.util.VIPConfig;
+
+public class VIPCfgFactory {
+    /**
+     * A lazy-loaded singleton VIPCfg using the "initialization-on-demand holder" design pattern
+     */
+    private static class GlobalCfgHolder {
+        static final VIPCfgWrapper globalCfg = new VIPCfgWrapper(new VIPCfg());
+    }
+
+    public static VIPCfgWrapper getGlobalCfg() {
+        return GlobalCfgHolder.globalCfg;
+    }
+
+    /**
+     * @deprecated This method was added for backwards compatibility with deprecated {@link VIPCfg#getInstance() getInstance}.
+     * Use the {@link #getGlobalCfg() getGlobalCfg} method.
+     */
+    public static VIPCfgWrapper getCfg() {
+        VIPCfgWrapper vipCfgWrapper = GlobalCfgHolder.globalCfg;
+
+        return GlobalCfgHolder.globalCfg;
+    }
+
+    /**
+     * A lazy-loaded singleton Map<String, VIPCfg> configs using the "initialization-on-demand holder" design pattern
+     */
+    private static class ConfigsHolder {
+        static final Map<String, VIPCfgWrapper> configs = new ConcurrentHashMap<>();
+    }
+
+    public static VIPCfgWrapper getCfg(String productName) {
+        if (!contains(productName)) {
+            synchronized (ConfigsHolder.configs) {
+                if (!contains(productName)) {
+                    VIPCfgWrapper cfgWrapper = new VIPCfgWrapper(new VIPCfg());
+                    cfgWrapper.setProductName(productName);
+                    addCfg(cfgWrapper);
+                }
+            }
+        }
+        return ConfigsHolder.configs.get(productName);
+    }
+
+    /**
+     * @deprecated Use {@link #getCfg(String) getCfg} instead.
+     * This method was added for backwards compatibility with deprecated {@link VIPCfg#getInstance()} and {@link VIPCfg#setProductName(String)} methods.
+     */
+    static void addCfg(String productName, VIPCfg cfg) {
```

Comment:
You can remove this directly because it's new.

Reply:
See comment section. Removing this will cause backwards compatibility issues for code that calls VIPCfg.getInstance.

Reply:
Removed!